kt-android-runtime/runtime/src/main/java/io/sdkgen/runtime/SdkgenHttpClient.kt

```diff
@@ -0,0 +1,229 @@
+            language = "nn"
+            region = "NO"
+            variant = ""
+        }
+
+        if (language.isEmpty() || !language.matches("\\\\[a-zA-Z]{2,8}".toRegex())) {
+            language = "und"
+        } else if (language == "iw") {
+            language = "he"
+        } else if (language == "in") {
+            language = "id"
+        } else if (language == "ji") {
+            language = "yi"
+        }
+
+        if (!region.matches("\\\\[a-zA-Z]{2}|\\\\[0-9]{3}".toRegex())) {
+            region = ""
+        }
+
+        if (!variant.matches("\\\\[a-zA-Z0-9]{5,8}|\\\\[0-9]\\\\[a-zA-Z0-9]{3}".toRegex())) {
+            variant = ""
+        }
+
+        val bcp47Tag = StringBuilder(language)
+        if (region.isNotEmpty()) {
+            bcp47Tag.append(sep).append(region)
+        }
+        if (variant.isNotEmpty()) {
+            bcp47Tag.append(sep).append(variant)
+        }
+
+        return bcp47Tag.toString()
+    }
+
+    @SuppressLint("HardwareIds")
+    private fun makeDeviceObj(): JsonObject {
+        return JsonObject().apply {
+            addProperty("type", "android")
+            addProperty("fingerprint", Settings.Secure.getString(applicationContext.contentResolver, Settings.Secure.ANDROID_ID))
+            add("platform", JsonObject().apply {
+                addProperty("version", Build.VERSION.RELEASE)
+                addProperty("sdkVersion", Build.VERSION.SDK_INT)
+                addProperty("brand", Build.BRAND)
+                addProperty("model", Build.MODEL)
+            })
+
+            try {
+                addProperty("version", applicationContext.packageManager.getPackageInfo(applicationContext.packageName, 0).versionName)
+            } catch (e: Exception) {
+                addProperty("version", "unknown")
+            }
+
+            addProperty("language", language())
+            add("screen", JsonObject().apply {
+                val manager = applicationContext.getSystemService(Context.WINDOW_SERVICE) as WindowManager
+                val display = manager.defaultDisplay
+                val size = Point()
+                display.getSize(size)
+                addProperty("width", size.x)
+                addProperty("height", size.y)
+            })
+
+            val pref = applicationContext.getSharedPreferences("api", Context.MODE_PRIVATE)
+            if (pref.contains("deviceId"))
+                addProperty("id", pref.getString("deviceId", null))
+        }
+    }
+
+    suspend fun makeRequest(functionName: String, bodyArgs: JsonObject?, timeoutMillis: Long = defaultTimeoutMillis): InternalResposne = suspendCoroutine { continuation ->
+        val httpTimer = Timer()
+        try {
+            val body = JsonObject().apply {
+                addProperty("id", callId())
+                add("device", makeDeviceObj())
+                addProperty("name", functionName)
+                add("args", bodyArgs ?: JsonObject())
+            }
```

Comment:
The body has changed a bit, [check dart runtime](https://github.com/sdkgen/sdkgen/blob/14ce542716afa427f0cfe2f62ab58a2f07ac2779/dart-runtime/lib/http_client.dart#L89-L107).